src/Microsoft.Azure.WebJobs.Host/Config/FluentBindingRule.cs

```diff
@@ -106,6 +108,43 @@ public FluentBindingRule<TAttribute> WhenIsNotNull(string propertyName)
             return this;
         }
 
+        /// <summary>
+        /// The subsequent Bind* operations only apply when the Attribute's property is not null. 
+        /// </summary>
+        /// <param name="propertyName"></param>
+        /// <returns></returns>
+        public FluentBindingRule<TAttribute> When<TValue>(string propertyName, TValue expected)
+        {
+            var prop = ResolveProperty(propertyName);
+
+            if (!typeof(TValue).IsEnum)
```

Comment:
I don't understand where in the method definition it's clear that TValue must be an enum? Perhaps you need a generic constraint or a different method name or something?

Reply:
C# doesn't allow generic constraints on Enums. :(   I can make the method more explicit. 

---
In reply to: [151524466](https://github.com/Azure/azure-webjobs-sdk/pull/1422#discussion_r151524466) [](ancestors = 151524466)